test/user.integration.js

```diff
@@ -0,0 +1,166 @@
+/*jshint -W030 */
+var loopback = require('../');
+var lt = require('loopback-testing');
+var path = require('path');
+var SIMPLE_APP = path.join(__dirname, 'fixtures', 'user-integration-app');
+var app = require(path.join(SIMPLE_APP, 'server/server.js'));
+var expect = require('chai').expect;
+
+describe('users - integration', function() {
+
+  lt.beforeEach.withApp(app);
+
+  before(function(done) {
+    // HACK: [rfeng] We have to reset the relations as they are polluted by
+    // other tests
+    app.models.User.hasMany(app.models.post);
+    app.models.User.hasMany(app.models.AccessToken,
+      {options: {disableInclude: true}});
+    app.models.AccessToken.belongsTo(app.models.User);
+    app.models.User.destroyAll(function(err) {
+      if (err) return done(err);
+      app.models.post.destroyAll(function(err) {
+        if (err) return done(err);
+        app.models.blog.destroyAll(function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('base-user', function() {
+    var userId;
+    var accessToken;
+
+    it('should create a new user', function(done) {
+      var url = '/api/users';
+
+      this.post(url)
+        .send({username: 'x', email: 'x@y.com', password: 'x'})
+        .expect(200, function(err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res.body.id).to.exist;
+          userId = res.body.id;
+          done();
+        });
+    });
+
+    it('should log into the user', function(done) {
+      var url = '/api/users/login';
+
+      this.post(url)
+        .send({username: 'x', email: 'x@y.com', password: 'x'})
+        .expect(200, function(err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res.body.id).to.exist;
+          accessToken = res.body.id;
+          done();
+        });
+    });
+
+    it('should create post for a given user', function(done) {
+      var url = '/api/users/' + userId + '/posts?access_token=' + accessToken;
+      this.post(url)
+        .send({title: 'T1', content: 'C1'})
+        .expect(200, function(err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res.body.title).to.be.eql('T1');
+          expect(res.body.content).to.be.eql('C1');
+          expect(res.body.userId).to.be.eql(userId);
+          done();
+        });
+    });
+
+    // FIXME: [rfeng] The test is passing if run alone. But it fails with
+    // `npm test` as the loopback models are polluted by other tests
+    it('should prevent access tokens from being included', function(done) {
```

Comment:
It does verify with the User/AccessToken relation setup (disableInclude = true), the accessTokens cannot be included via a user, even though the pollution from other tests forces me to use the hack to restore the relations to mimic the original configuration from the test fixture.

I'm open to better ways to test the feature but it's better not to let the security hole drags too long.